apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/utility/DataUtilityCommons.java

```diff
@@ -71,27 +73,31 @@ public static void getFDADrugCodes(String outputDir, String fdaFile)
    *
    * @param convertedNdcDataFile the output file/resource to produce.
    * @param workingDir a directory that temporary/working files can be written to.
+   * @param version The BFD version.
    * @throws IOException (any errors encountered will be bubbled up).
    */
-  public static void buildProductsResource(Path convertedNdcDataFile, Path workingDir)
+  public static void buildProductsResource(
+      Path convertedNdcDataFile, Path workingDir, String version)
       throws IOException, IllegalStateException {
     // download FDA NDC file
     Path downloadedNdcZipFile =
         Paths.get(workingDir.resolve("ndctext.zip").toFile().getAbsolutePath());
+    URL ndctextZipUrl = new URL("https://www.accessdata.fda.gov/cder/ndctext.zip");
     if (!Files.isReadable(downloadedNdcZipFile)) {
-      ClassLoader classLoader = DataUtilityCommons.class.getClassLoader();
-      File ndcFile =
-          new File(Objects.requireNonNull(classLoader.getResource("ndctext.zip")).getFile());
-      FileUtils.copyFile(ndcFile, new File(downloadedNdcZipFile.toFile().getAbsolutePath()));
+      HttpURLConnection connection = (HttpURLConnection) ndctextZipUrl.openConnection();
+      connection.setRequestProperty(
+          "User-Agent", String.format("BFD / %s (Beneficiary FHIR Data Server)", version));
```

Comment:
Based on the [examples](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent), I think we should remove the spaces, but this otherwise looks good. Thanks!

```suggestion
          "User-Agent", String.format("BFD/%s (Beneficiary FHIR Data Server)", version));
```